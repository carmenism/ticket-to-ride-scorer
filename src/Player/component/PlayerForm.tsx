
import * as React from 'react';
import { PlayerScore } from '../model';
import Train from './Train';

export interface PlayerFormProps extends PlayerScore {
  handleChange: (event: any) => void;
}

const PlayerForm: React.SFC<PlayerFormProps> = (props) => {
  const trains: any[] = [
    {
      length: 1,
      value: props.numberTrainsLength1,
      name: 'numberTrainsLength1'
    }, {
      length: 2,
      value: props.numberTrainsLength2,
      name: 'numberTrainsLength2'
    }, {
      length: 3,
      value: props.numberTrainsLength3,
      name: 'numberTrainsLength3'
    }, {
      length: 4,
      value: props.numberTrainsLength4,
      name: 'numberTrainsLength4'
    }, {
      length: 5,
      value: props.numberTrainsLength5,
      name: 'numberTrainsLength5'
    }, {
      length: 6,
      value: props.numberTrainsLength6,
      name: 'numberTrainsLength6'
    }, {
      length: 8,
      value: props.numberTrainsLength8,
      name: 'numberTrainsLength8'
    },
  ];

  return (
    <form onSubmit={this.handleSubmit}>
      {
        trains.map((train, i) => 
          <Train
            handleChange={props.handleChange}
            length={train.length}
            name={train.name}
            value={train.value}
            key={train.name}
          />
        )
      }
    </form>
  );
};

export default PlayerForm;
