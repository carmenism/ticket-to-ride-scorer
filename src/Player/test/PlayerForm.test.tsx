import * as React from 'react';
import { shallow } from 'enzyme';
import PlayerForm from '../component/PlayerForm';

describe('PlayerForm', () => {
  let wrapper: any;
  const handleChange = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <PlayerForm
        numberTrainsLength1={1}
        numberTrainsLength2={3}
        numberTrainsLength3={2}
        numberTrainsLength4={0}
        numberTrainsLength5={1}
        numberTrainsLength6={0}
        numberTrainsLength8={0}
        routesAchieved={[]}
        routesMissed={[]}
        numberUnusedStations={3}
        longestRoute={false}
        handleChange={handleChange}
      />
    );
  });

  it('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
