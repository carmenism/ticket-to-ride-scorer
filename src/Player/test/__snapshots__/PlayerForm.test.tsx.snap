// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerForm Matches snapshot 1`] = `
<form>
  <PlayerForm
    handleChange={[MockFunction]}
    key="numberTrainsLength1"
    length={1}
    name="numberTrainsLength1"
    value={1}
  />
  <PlayerForm
    handleChange={[MockFunction]}
    key="numberTrainsLength2"
    length={2}
    name="numberTrainsLength2"
    value={3}
  />
  <PlayerForm
    handleChange={[MockFunction]}
    key="numberTrainsLength3"
    length={3}
    name="numberTrainsLength3"
    value={2}
  />
  <PlayerForm
    handleChange={[MockFunction]}
    key="numberTrainsLength4"
    length={4}
    name="numberTrainsLength4"
    value={0}
  />
  <PlayerForm
    handleChange={[MockFunction]}
    key="numberTrainsLength5"
    length={5}
    name="numberTrainsLength5"
    value={1}
  />
  <PlayerForm
    handleChange={[MockFunction]}
    key="numberTrainsLength6"
    length={6}
    name="numberTrainsLength6"
    value={0}
  />
  <PlayerForm
    handleChange={[MockFunction]}
    key="numberTrainsLength8"
    length={8}
    name="numberTrainsLength8"
    value={0}
  />
</form>
`;
